client/packages/system/src/Manage/Campaigns/ListView/ListView.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import {
+  TableProvider,
+  DataTable,
+  useColumns,
+  createTableStore,
+  NothingHere,
+  useUrlQueryParams,
+  useEditModal,
+  useTranslation,
+  ColumnFormat,
+  CampaignNode,
+  GenericColumnKey,
+  useTableStore,
+  useDeleteConfirmation,
+  useNotification,
+} from '@openmsupply-client/common';
+import { Footer } from './Footer';
+import { CampaignEditModal } from './CampaignEditModal';
+import { AppBarButtons } from './AppBarButtons';
+import { CampaignRowFragment, DraftCampaign, useCampaigns } from '../api';
+
+const CampaignsComponent = () => {
+  const t = useTranslation();
+  const {
+    // filter,
+    updateSortQuery,
+    updatePaginationQuery,
+    queryParams: { sortBy, page, first, offset, filterBy },
+  } = useUrlQueryParams({ initialSort: { key: 'name', dir: 'asc' } });
+
+  const queryParams = { sortBy, first, offset, filterBy };
+  const {
+    query: { data, isError, isLoading },
+    upsert: { upsert },
+    delete: { deleteCampaign },
+    draft,
+    updateDraft,
+  } = useCampaigns(queryParams);
+
+  const pagination = { page, first, offset };
+
+  const { isOpen, onClose, onOpen } = useEditModal();
+
+  const { error, success } = useNotification();
+
+  const { selectedRows } = useTableStore(state => ({
+    selectedRows: Object.keys(state.rowState)
+      .filter(id => state.rowState[id]?.isSelected)
+      .map(selectedId => data?.nodes?.find(({ id }) => selectedId === id))
+      .filter(Boolean),
+  }));
+
+  const save = async () => {
+    const result = await upsert();
+    if (result?.__typename === 'CampaignNode')
+      success(t('messages.campaign-saved'))();
+    else if ('error' in result)
+      error(
+        `${t('messages.error-saving-campaign')} — ${result?.error?.description ?? ''}`
+      )();
+  };
+
+  const confirmAndDelete = useDeleteConfirmation({
+    selectedRows,
+    deleteAction: async () =>
+      selectedRows.forEach(row => {
+        if (row) {
+          deleteCampaign(row.id);
+        }
```

Comment:
I don't like doing a separate mutation for each deletion, but the current GraphQL Delete endpoint only accepts a single ID. We should probably change this to accept an array of IDs like some of the other ones. New issue you reckon?

Reply:
Seems very reasonable to me to loop through for now. I can't imagine mass campaign deleting occurring other than from testing?

@jmbrunskill do you have any thoughts?

Reply:
Actually seems fine - we use the same looping in a number of other parts of the app (asset delete, or returns rows for example). And this seems less likely to be an area of the app with high traffic.

Reply:
However this will give a confirmation of all successfully deleted if one or more of the rows are successfully deleted.


https://github.com/user-attachments/assets/dcdca105-9b1d-47fe-b335-7cbfcf52806b

But I think you can capture success or failure in bulk with the bulk delete patterns like:

```
  const deleteAction = async () => {
    await Promise.all(selectedRows.map(row => mutateAsync(row.id)))
      .then(() => queryClient.invalidateQueries(api.keys.base()))
      .catch(err => {
        throw err;
      });
  };
  ```